fMRISurface/scripts/SubcorticalProcessing.sh

```diff
@@ -30,50 +30,50 @@ echo "${script_name}: VolumefMRI: ${VolumefMRI}"
+    ${CARET7DIR}/wb_command -cifti-create-dense-timeseries ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii -volume "$VolumefMRI".nii.gz "$ROIFolder"/ROIs."$BrainOrdinatesResolution".nii.gz
+else
+    echo "${script_name}: Creating subject-roi subcortical cifti at differing fMRI resolution"
+    ${CARET7DIR}/wb_command -volume-affine-resample "$ROIFolder"/ROIs."$BrainOrdinatesResolution".nii.gz $FSLDIR/etc/flirtsch/ident.mat "$VolumefMRI".nii.gz ENCLOSING_VOXEL "$ResultsFolder"/ROIs."$FinalfMRIResolution".nii.gz
+    ${CARET7DIR}/wb_command -cifti-create-dense-timeseries ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii -volume "$VolumefMRI".nii.gz "$ResultsFolder"/ROIs."$FinalfMRIResolution".nii.gz
+    rm -f "$ResultsFolder"/ROIs."$FinalfMRIResolution".nii.gz
+fi
 
-if [ 1 -eq `echo "$BrainOrdinatesResolution == $FinalfMRIResolution" | bc -l` ] ; then
-	echo "${script_name}: Doing volume parcel resampling without first applying warp"
-	${CARET7DIR}/wb_command -volume-parcel-resampling "$VolumefMRI".nii.gz "$ROIFolder"/ROIs."$BrainOrdinatesResolution".nii.gz "$ROIFolder"/Atlas_ROIs."$BrainOrdinatesResolution".nii.gz $Sigma "$VolumefMRI"_AtlasSubcortical_s"$SmoothingFWHM".nii.gz -fix-zeros
+echo "${script_name}: Dilating out zeros"
+#dilate out any exact zeros in the input data, for instance if the brain mask is wrong
+${CARET7DIR}/wb_command -cifti-dilate ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii COLUMN 0 10 ${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii
+rm -f ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii
+
+echo "${script_name}: Generate atlas subcortical template cifti"
+${CARET7DIR}/wb_command -cifti-create-label ${ResultsFolder}/${NameOffMRI}_temp_template.dlabel.nii -volume "$ROIFolder"/Atlas_ROIs."$BrainOrdinatesResolution".nii.gz "$ROIFolder"/Atlas_ROIs."$BrainOrdinatesResolution".nii.gz
+
+if [[ `echo "${Sigma} > 0" | bc -l | cut -f1 -d.` == "1" ]]
+then
+    echo "${script_name}: Smoothing and resampling"
+    #this is the whole timeseries, so don't overwrite, in order to allow on-disk writing, then delete temporary
+    ${CARET7DIR}/wb_command -cifti-smoothing ${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii 0 ${Sigma} COLUMN ${ResultsFolder}/${NameOffMRI}_temp_subject_smooth.dtseries.nii -fix-zeros-volume
+    #resample, delete temporary
+    ${CARET7DIR}/wb_command -cifti-resample ${ResultsFolder}/${NameOffMRI}_temp_subject_smooth.dtseries.nii COLUMN ${ResultsFolder}/${NameOffMRI}_temp_template.dlabel.nii COLUMN ADAP_BARY_AREA CUBIC ${ResultsFolder}/${NameOffMRI}_temp_atlas.dtseries.nii -volume-predilate 10
+    rm -f ${ResultsFolder}/${NameOffMRI}_temp_subject_smooth.dtseries.nii
 else
-	echo "${script_name}: Creating subcortical ROI volume in original fMRI resolution"
-	cp "$GrayordinatesSpaceDIR"/Atlas_ROIs."$FinalfMRIResolution".nii.gz "$ROIFolder"/Atlas_ROIs."$FinalfMRIResolution".nii.gz
```

Comment:
Major difference: keith's version uses the unset variable $GrayordinatesSpaceDIR (it is used in postfreesurfer, but not in generic surface), which causes keith's version to error.  My version doesn't use this variable, but instead expects the atlas ROIs to have already been copied to the subject folder, like the original script (this copy happens in postfreesurfer, scripts/FreeSurfer2Caret...).